privacyscore/backend/management/commands/scanfromfile.py

```diff
@@ -28,22 +28,33 @@ class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument('file_path')
         parser.add_argument('-s', '--sleep-between-scans', type=float, default=0)
+        parser.add_argument('-c', '--create-list-name')
 
     def handle(self, *args, **options):
         if not os.path.isfile(options['file_path']):
             raise ValueError('file does not exist!')
-        self.stdout.write('Reading from file {}'.format(options['file_path']))
+        scan_list = None
+
+        if options['create_list_name']:
+            if ScanList.objects.filter(name=options['create_list_name']).exists():
+                raise ValueError('Scan List already exists!')
 
-        start = timezone.now()
+            self.stdout.write('Creating ScanList {}'.format(options['create_list_name']))
+            scan_list = ScanList.objects.create(name=options['create_list_name'])
+            scan_list.private = True
+            scan_list.save()
 
+        self.stdout.write('Reading from file {}'.format(options['file_path']))
+        sites = []
```

Comment:
you're now allowing the same site multiple times by changing the set to a list. Is that intended?  Otherwise make it a `set()`.

Reply:
That is actually not true. In the for loop below, I am doing `site = Site.objects.get_or_create(url=url)[0]` which makes sure same site is not created multiple times.

Reply:
Good catch!
It's still multiple times in the list, though. And thus we call `scan` multiple times on that site. It's probably not causing harm, because the Site is clever enough to return COOLDOWN. And the ScanList is hopefully clever enough to not contain a site multiple times.

So yeah, this is probably good enough.